src/Utility/ExprFunctions.C

```diff
@@ -398,15 +398,14 @@ void EvalFunctions::addDerivative (const std::string& functions,
 }
 
 
-template<>
-Vec3 VecFuncExpr::evaluate (const Vec3& X) const
+template <class ParentFunc, class Ret>
+Ret EvalMultiFunction<ParentFunc,Ret>::evaluate (const Vec3& X) const
 {
-  Vec3 result;
-
-  for (size_t i = 0; i < 3 && i < nsd; ++i)
-    result[i] = (*p[i])(X);
+  std::vector<Real> res_array(this->p.size());
+  for (size_t i = 0; i < this->p.size(); ++i)
+    res_array[i] = (*this->p[i])(X);
 
-  return result;
+  return Ret(res_array);
```

Comment:
Will this work for SymmTensor for Axi-Symmetrix problems, where a second bool value is fed to its' constructor (see comment below)?

Reply:
yes.